### PR TITLE
Add messaging to retrieve the JS API version supported by the extension

### PR DESCRIPTION
**Pull request by iuliaion**
When the page first loads (and on any explicit user refresh), the u2fdemo.js sends a message to the extension, finds out and stores the JS API version.

Later, when calling register or sign, the page can decide how to form the request based on which API version is supported by the extension.

For old extensions who do not understand the get JS API message, the JS API version will be set to undefined.

### CODE REVIEW

File: u2f-chrome-extension/messagetypes.js [hunk 1]
```diff
@@ -18,5 +18,7 @@ var MessageTypes = {
   U2F_REGISTER_REQUEST: 'u2f_register_request',
   U2F_SIGN_REQUEST: 'u2f_sign_request',
   U2F_REGISTER_RESPONSE: 'u2f_register_response',
-  U2F_SIGN_RESPONSE: 'u2f_sign_response'
+  U2F_SIGN_RESPONSE: 'u2f_sign_response',
+  U2F_GET_API_VERSION_REQUEST: 'u2f_get_api_version_request',
```
Comment (leshi) [inline u2f-chrome-extension/messagetypes.js]: can you separate the addition of the getApiVersion() function into one CL and the rest of the modifications into another CL?

Comment (leshi) [inline u2f-chrome-extension/messagetypes.js]: Specifically, can you separate the addition of get_api_version_request from the addition of  u2f_sign_response

Comment (iuliaion) [inline u2f-chrome-extension/messagetypes.js]: I am not adding u2f_sign_response. The diff is just weird cause I put a coma at the end of that line.

Comment (leshi) [inline u2f-chrome-extension/messagetypes.js]: Right.

I'm not a fan of this diff tool -- it shows the minuses in the left screen and the pluses in the right one.

File: u2f-chrome-extension/u2f-api.js [hunk 2]
```diff
@@ -344,7 +353,7 @@ u2f.responseHandler_ = function(message) {
  */
 u2f.sign = function(signRequests, callback, opt_timeoutSeconds) {
   u2f.getPortSingleton_(function(port) {
-    var reqId = ++u2f.reqCounter_;
+  var reqId = ++u2f.reqCounter_;
```
Comment (leshi) [inline u2f-chrome-extension/u2f-api.js]: why did you delete the indent here?

Comment (iuliaion) [inline u2f-chrome-extension/u2f-api.js]: Ah, just mistake. Will fix it.

Comment (iuliaion) [inline u2f-chrome-extension/u2f-api.js]: Done.

File: u2f-chrome-extension/approvedorigins.js [hunk 3]
```diff
@@ -1,13 +1,8 @@
...
-//
-// Use of this source code is governed by a BSD-style
-// license that can be found in the LICENSE file or at
-// https://developers.google.com/open-source/licenses/bsd
-
 /**
  * @fileoverview Provides an interface to check whether the user has approved
  * an origin to use security keys.
  *
+ * @author juanlang@google.com (Juan Lang)
```
Comment (juanlang) [inline u2f-chrome-extension/approvedorigins.js]: I think these should have been removed by the export tool? I'm okay with my name being on these files, but it seems like the export might not have worked. See below.

File: u2f-chrome-extension/b64.js [hunk 4]
```diff
@@ -1,10 +1,5 @@
-// Copyright 2014 Google Inc. All rights reserved
-//
-// Use of this source code is governed by a BSD-style
-// license that can be found in the LICENSE file or at
-// https://developers.google.com/open-source/licenses/bsd
-
 // WebSafeBase64Escape and Unescape.
+// GOOGLE-INTERNAL mschilder@google.com
```
Comment (juanlang) [inline u2f-chrome-extension/b64.js]: This definitely shouldn't be here.

File: u2f-chrome-extension/debug.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
+/**
```
Comment (juanlang) [inline u2f-chrome-extension/debug.js]: This points out a bug in the upstream source, this really has no reason to belong here. I'll send a cl to fix, then perhaps you could re-export.

File: u2f-chrome-extension/gnubby-u2f.js [hunk 6]
```diff
@@ -1,15 +1,12 @@
...
-// license that can be found in the LICENSE file or at
-// https://developers.google.com/open-source/licenses/bsd
-
 /**
  * @fileoverview Gnubby methods related to U2F support.
  */
 'use strict';
 
 // Commands and flags of the Gnubby applet
+// BEGIN GOOGLE-INTERNAL
```
Comment (juanlang) [inline u2f-chrome-extension/gnubby-u2f.js]: Once again, this definitely shouldn't be here.
